source/aws-bootstrap-kit/lib/account.ts

```diff
@@ -84,77 +84,132 @@ export class Account extends Construct {
+      const describeAccount = {
+        service: "Organizations",
+        action: "describeAccount",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse(
+          "Account.Id"
+        ),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+        parameters: {
+          AccountId: accountId
+        }
+      };
+
+      account = new cr.AwsCustomResource(this, "ExistingAccountCustomResource", {
+        onCreate: describeAccount,
+        onUpdate: describeAccount,
+        onDelete: describeAccount,
+        policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+          resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+        })
+      });
+
+      accountProps.name = account.getResponseField("Account.Name");
+      accountProps.email = account.getResponseField("Account.Email");
+    }
+
+    // add tags
+    const tags: { Key: string; Value: any; }[] = [];
+    tags.push({Key: 'Email', Value: accountProps.email});
+    tags.push({Key: 'AccountName', Value: accountProps.name});
+    if (accountProps.type != null) tags.push({Key: 'AccountType', Value: accountProps.type});
+    if (accountProps.stageName != null) tags.push({Key: 'StageName', Value: accountProps.stageName});
+    if (accountProps.stageOrder != null) tags.push({Key: 'StageOrder', Value: accountProps.stageOrder.toString()});
+    if (hostedServices != null) tags.push({Key: 'HostedServices', Value: hostedServices});
+    const tagAccount = {
+      service: "Organizations",
+      action: "tagResource",
+      region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      physicalResourceId: cr.PhysicalResourceId.of(`tags-${accountId}`),
+      parameters: {
+        ResourceId: accountId,
+        Tags: tags
+      }
+    };
+    new cr.AwsCustomResource(this, "TagAccountCustomResource", {
+      onCreate: tagAccount,
+      onUpdate: tagAccount,
+      onDelete: {
+        service: "Organizations",
+        action: "untagResource",
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+        parameters: {
+          ResourceId: accountId,
+          TagKeys: tags.map(t => t.Key)
+        }
+      },
+      policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+        resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+      })
+    });
 
-    accountProps.id = accountId;
     this.accountName = accountProps.name;
     this.accountId = accountId;
     this.accountStageName = accountProps.stageName;
+    this.accountStageOrder = accountProps.stageOrder;
 
-    new ssm.StringParameter(this, `${accountProps.name}-AccountDetails`, {
+    let ssmParam = new ssm.StringParameter(this, exitingAccount?`${accountId}-AccountDetails`:`${accountProps.name}-AccountDetails`, {
       description: `Details of ${accountProps.name}`,
       parameterName: `/accounts/${accountProps.name}`,
```

Comment:
Can you double check accountName regex vs. ssm parameterName one

Reply:
account id is not resolved when creating an account.
account name is not resolved when reusing an account.
That's why I used that. I remember having the issue in both case... 
That's just the id of the resource, not so impactful.

parameterName regex: a-zA-Z0-9_.- and /
accountName regex: [\s\S]* ("any of the characters in the ASCII character range") ==> may be an issue